### PR TITLE
Added endpoints for animal_services table

### PR DESCRIPTION
**Pull request by tanveersaleem786**
Created endpoints for animal_services table that will be used at front end

### CODE REVIEW

**Review by wilcoxva — COMMENTED** [review]
looks good!

File: api/animal_service/animalServiceRouter.js [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+ *              type: object
+ *              properties:
+ *                message:
+ *                  type: string
+ *                  description: A message about the result
+ *                  example: Animal service update
+ *                animal:
+ *                  $ref: '#/components/schemas/AnimalService'
+ */
+router.put(
```
Comment (TylerAlsop) [inline api/animal_service/animalServiceRouter.js:249]: Wow good job keeping track of all of the authentication and items to update!

Comment (TylerAlsop) [inline api/animal_service/animalServiceRouter.js:249]: Wow good job keeping track of all of the authentication and items to update!

File: api/app.js [hunk 2]
```diff
@@ -23,12 +23,11 @@ const indexRouter = require('./index/indexRouter');
...
 const animalRouter = require('./animal/animalRouter');
 const serviceRouter = require('./service/serviceRouter');
+const animalServiceRouter = require('./animal_service/animalServiceRouter');
 
 const app = express();
 
-app.use(
-  cors()
-);
+app.use(cors());
```
Comment (TylerAlsop) [inline api/app.js:30]: Nice fix Tommy!

File: api/animal_service/animalServiceModel.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
+const db = require('../../data/db-config');
+
+const findAll = async () => {
+  return await db('animal_services as ans')
+    .join('services as s', 's.id', 'ans.service_id')
```
Comment (TylerAlsop) [inline api/animal_service/animalServiceModel.js:5]: Good job with your join statements in this file. They can be hard to keep track of.

File: api/middleware/animalService.js [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+  isAnimalServiceUnique: () => {
+    return async (req, res, next) => {
+      try {
+        const { animal_id, service_id } = req.body;
+        const service_qry = db('animal_services')
+          .where('animal_id', animal_id)
+          .where('service_id', service_id);
+
+        if (await service_qry.first()) {
+          res.status(404).json({ message: 'Animal Service alreday exist' });
```
Comment (TylerAlsop) [inline api/middleware/animalService.js:105]: Minor spelling error. Should read, 'Animal Service already exists.'
We can probably fix this after merging.

**Review by TylerAlsop — APPROVED** [review]
Everything looks good to me. Just a minor spelling error in a response message in api/middleware/animalService.js. I'll probably checkout a new branch to fix the spelling errors in these last two pull-requests.

**Pull request merged**